content/resources/crownlabs_ssh/crownlabs_ssh.md

```diff
@@ -0,0 +1,97 @@
++++
+fragment = "content"
+#disabled = true
+date = "2021-11-12"
+weight = 100
+# title = "Connecting to CrownLabs with SSH"
++++
+
+## Introduction
+
+[**CrownLabs**](https://crownlabs.polito.it/) allows users access via SSH to their VMs using SSH directly on their machines.
+
+In this guide you are going to learn how to generate a key on the main operative systems (Windows, Linux, MacOS), how to load it on [**CrownLabs**](https://crownlabs.polito.it/) and log in your [**CrownLabs**](https://crownlabs.polito.it/)'VM using SSH from terminal.
+
+## How to generate an SSH key pair
+
+1. Open a terminal and run the following command to create a [**RSA**](<https://it.wikipedia.org/wiki/RSA_(crittografia)>) key pair :
+
+   ```
+   ssh-keygen -t rsa
+   ```
+
+   You can use the `-b` option to specify the length (bit size) of the key, as shown in the following example:
+
+   ```
+   ssh-keygen -b 2048 -t rsa
+   ```
+
+2. The command will prompt you to enter the path to the file where you want to save the key. A default path and file name similar to `/home/user_name/.ssh/id_rsa` (`C:\Users\username\.ssh\id_rsa` on **windows**) are suggested in parentheses. To accept the default path and file name (_suggested_), press **Enter**. Otherwise, enter the required path and file name, and then press **Enter**. Beware that from now on, this tutorial will assume that you have chosen the default settings.
+
+3. A password will be asked. You are free to insert one or to ignore it.
+
+Now you will find in the previously chosen path (`$HOME/.ssh` if you chose the default path, `C:\Users\yourusername\.ssh`) a file with a **.pub** extension (if you chose the default file name you will find `id_rsa.pub`). This is the file containing your **SSH public key**.
+
+Print the content of your **SSH public key** running:
+
+`cat $HOME/.ssh/id_rsa.pub` or opening `C:\Users\yourusername\.ssh` with an editor in **windows**
+
+Now copy the **SSH public key**.
```

Comment:
Would add a note that the other file contains the private key which must not be shared with anyone.